crates/download/src/main.rs

```diff
@@ -0,0 +1,446 @@
+
+    /// ダウンロードするacceleratorを指定する(cudaはlinuxのみ)
+    #[arg(value_enum, long, default_value(<&str>::from(Accelerator::default())))]
+    accelerator: Accelerator,
+
+    /// ダウンロードするcpuのアーキテクチャを指定する
+    #[arg(value_enum, long, default_value(CpuArch::default_opt().map(<&str>::from)))]
+    cpu_arch: CpuArch,
+
+    /// ダウンロードする対象のOSを指定する
+    #[arg(value_enum, long, default_value(Os::default_opt().map(<&str>::from)))]
+    os: Os,
+}
+
+#[derive(Default, ValueEnum, Display, IntoStaticStr, Clone, Copy, PartialEq)]
+#[strum(serialize_all = "kebab-case")]
+enum Accelerator {
+    #[default]
+    Cpu,
+    Cuda,
+    Directml,
+}
+
+#[derive(ValueEnum, Display, IntoStaticStr, Clone, Copy, PartialEq)]
+#[strum(serialize_all = "kebab-case")]
+enum CpuArch {
+    X86,
+    X64,
+    Aarch64,
+}
+
+impl CpuArch {
+    fn default_opt() -> Option<Self> {
+        match env::consts::ARCH {
+            "x86_64" => Some(Self::X64),
+            "aarch64" => Some(Self::Aarch64),
+            _ => None,
+        }
+    }
+}
+
+#[derive(ValueEnum, Display, IntoStaticStr, Clone, Copy, PartialEq)]
+#[strum(serialize_all = "kebab-case")]
+enum Os {
+    Windows,
+    Linux,
+    Osx,
+}
+
+impl Os {
+    fn default_opt() -> Option<Self> {
+        match env::consts::OS {
+            "windows" => Some(Self::Windows),
+            "linux" => Some(Self::Linux),
+            "macos" => Some(Self::Osx),
+            _ => None,
+        }
+    }
+}
+
+#[tokio::main(flavor = "current_thread")]
+async fn main() -> anyhow::Result<()> {
+    setup_logger();
+
+    let Args {
+        min,
+        output,
+        version,
+        additional_libraries_version,
+        accelerator,
+        cpu_arch,
+        os,
+    } = Args::parse();
+
+    let octocrab = &octocrab()?;
+
+    let core = find_gh_asset(octocrab, CORE_REPO_NAME, &version, |tag| {
+        let cpu_arch = match (os, cpu_arch) {
+            (Os::Linux, CpuArch::Aarch64) => "arm64",
+            (_, cpu_arch) => cpu_arch.into(),
+        };
+        let accelerator = match (os, accelerator) {
+            (Os::Linux, Accelerator::Cuda) => "gpu",
+            (_, accelerator) => accelerator.into(),
+        };
+        format!("{CORE_REPO_NAME}-{os}-{cpu_arch}-{accelerator}-{tag}.zip")
+    })
+    .await?;
+
+    let additional_libraries = if accelerator != Accelerator::Cpu {
+        Some(
+            find_gh_asset(
+                octocrab,
+                ADDITIONAL_LIBRARIES_REPO_NAME,
+                &additional_libraries_version,
+                |_| {
+                    let accelerator = match accelerator {
+                        Accelerator::Cpu => unreachable!(),
+                        Accelerator::Cuda => "CUDA",
+                        Accelerator::Directml => "DirectML",
+                    };
+                    format!("{accelerator}-{os}-{cpu_arch}.zip")
+                },
+            )
+            .await?,
+        )
+    } else {
+        None
+    };
```

Comment:
一応こんな書き方もできますが、うーん見やすい?

```suggestion
    use futures_util::future::OptionFuture;

    let additional_libraries = OptionFuture::from((accelerator != Accelerator::Cpu).then(|| {
        find_gh_asset(
            octocrab,
            ADDITIONAL_LIBRARIES_REPO_NAME,
            &additional_libraries_version,
            |_| {
                let accelerator = match accelerator {
                    Accelerator::Cpu => unreachable!(),
                    Accelerator::Cuda => "CUDA",
                    Accelerator::Directml => "DirectML",
                };
                format!("{accelerator}-{os}-{cpu_arch}.zip")
            },
        )
    }))
    .await
    .transpose()?;
```

Reply:
まあいいや。やっちゃいます。

`bool` → `Option<impl Future>` → `Option<Result<_, _>>` → `Result<Option<_>, _>`という直列の流れになっている方がわかりやすいかと思うので。